Automattic/jetpack
# eslint: Enable more jest linting

#### Changes proposed in this Pull Request:

This adds more linting rules for jest tests, to help make things more
consistent and to provide better messages on failures.

Of the added rules, the only one of note that needed fixes anywhere was
one preferring `toBe()` over `toEqual()` for primitive values, which
generates better messages on failure.

Also add linting rules for `@testing-library/react`.

Of the added rules, the only one that needed fixes anywhere was in an
Instant Search test where it wanted us to use `screen.getByRole()`
instead of calling `getByRole()` on the result of `render()`.

#### Jetpack product discussion


None

#### Does this pull request change what data or activity we track or use?


No

#### Testing instructions:






* Is CI happy?
* There are more rules for both [jest](https://github.com/jest-community/eslint-plugin-jest#rules) and [testing-library](https://github.com/testing-library/eslint-plugin-testing-library#supported-rules). Anyone think we should enable any of those?
  * I was tempted to add [jest/consistent-test-it](https://github.com/jest-community/eslint-plugin-jest/blob/main/docs/rules/consistent-test-it.md), but it looks like either direction would change about half the tests in the codebase so I left it out for now.
